### PR TITLE
Add tool and tests to lint shell files

### PR DESCRIPTION
**Pull request by ingydotnet**

### CODE REVIEW

File: tool/docker/bin/lint-shell [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+source "$YAML_SPEC_ROOT/tool/lib/runner.bash"
+
+check() (
+  file=$(command -v shellcheck)
+  [[ $file && -e $file ]] || return
+  [[ $($file --version) == *0.7.2* ]] || return
```
Comment (Thom1729) [inline tool/docker/bin/lint-shell:8]: Is it necessary to pin the version so precisely?

File: tool/docker/lint-shell/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/check-spec-file
```
Comment (Thom1729) [inline tool/docker/lint-shell/.gitignore:1]: What does this refer to?

File: tool/docker/lint-shell/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu:20.04
+
+RUN apt-get update \
+ && apt-get install -y \
+        apt-utils \
+        wget \
+        xz-utils \
+ && wget https://github.com/koalaman/shellcheck/releases/download/v0.7.2/shellcheck-v0.7.2.linux.x86_64.tar.xz \
```
Comment (Thom1729) [inline tool/docker/lint-shell/Dockerfile:8]: Not available via apt?

File: tool/bin/tex-to-img [hunk 4]
```diff
@@ -8,22 +8,24 @@ main() (
   for input; do
     output=${input%.tex}.svg
     output=${outdir%/}/${output#tex/}
-    tmp=/tmp/$(basename "${input%.tex}.pdf")
```
Comment (Thom1729) [inline tool/bin/tex-to-img:11]: Was this left over from something else?

File: tool/bin/lint-shell [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -e -u -o pipefail
+shopt -s inherit_errexit
+
+main() (
+  export SHELLCHECK_OPTS='--shell=bash --exclude=1091,2030,2031'
```
Comment (Thom1729) [inline tool/bin/lint-shell:7]: I assume there's no way to specify a config file for shellcheck?

File: tool/bin/tex-to-img (hunk 4 above)
Comment (ingydotnet) [reply tool/bin/tex-to-img:11]: Must have been. shellcheck flagged it as unused, and I concur.

File: tool/docker/bin/lint-shell (hunk 1 above)
Comment (ingydotnet) [reply tool/docker/bin/lint-shell:8]: Yes. That's the latest stable. If prior or future version was called, the tests would likely fail.
The docker fallback assures that the test always runs now.
BTW, I added top level `make test-docker` to invoke docker fallbacks for every spec util.

File: tool/docker/lint-shell/.gitignore (hunk 2 above)
Comment (ingydotnet) [reply tool/docker/lint-shell/.gitignore:1]: The should be `/lint-shell`. Thanks.

Comment (ingydotnet) [reply tool/docker/lint-shell/.gitignore]: Fixed in 8700ab1

File: tool/docker/lint-shell/Dockerfile (hunk 3 above)
Comment (ingydotnet) [reply tool/docker/lint-shell/Dockerfile:8]: ubuntu:20.04 has shellcheck v0.7.0. Want 0.7.2 latest.

File: tool/docker/lint-shell/.gitignore (hunk 2 above)
Comment (Thom1729) [reply tool/docker/lint-shell/.gitignore]: Why does it need to be ignored?

File: tool/bin/lint-shell (hunk 5 above)
Comment (ingydotnet) [reply tool/bin/lint-shell]: Good idea. Done in c696798

Comment (Thom1729) [reply tool/bin/lint-shell]: Do we still need the `lint-shell` file then? No strong opinion.

**Pull request merged**